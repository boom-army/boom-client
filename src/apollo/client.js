import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          users: offsetLimitPagination(),
        },
      },
    },
  }),
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
  },
  context: async ({ req }) => {
    const token = req?.headers?.authorization || null;
    const user= token ? JSON.parse(localStorage.getItem("user")) : null;
    return { user, isLoggedIn: !!token };
  },
});
