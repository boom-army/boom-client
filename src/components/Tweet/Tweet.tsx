import React from "react";
import "linkify-plugin-hashtag";
import "linkify-plugin-mention";
import Linkify from "linkify-react";
import UserAvatar from "../UserAvatar";
import moment from "moment";
import { CommentIcon } from "../Icons";
import { EmojiTweet, Retweet } from "./index";
import { ImageBox } from "../ImageBox";
import { LAMPORTS_PER_SOL } from "../../constants/math";
import { Link } from "react-router-dom";
import { List as ReactionsList } from "../Reactions/List";
import { NFTTweet } from "../NFT/NFTTweet";
import { TipCreator } from "../TipCreator";
import { styled } from "@mui/material/styles";
import { Tweet } from "../../generated/graphql";
import { VideoContainer } from "../Giphy/VideoContainer";
import { setDate } from "../../utils";
import { useReaction } from "../../hooks/useReaction";

const Wrapper = styled("div")((props) => ({
  display: "flex",
  borderBottom: `1px solid ${props.theme.tertiaryColor}`,
  padding: "1.5rem 1rem 1rem 1rem",

  ".avatar": {
    marginRight: "1em",
    marginBottom: "1em",
  },

  ".tweet-info": {
    width: "100%",
  },

  ".tweet-info-user": {
    display: "flex",
  },

  ".tweet-info-user span.username": {
    fontWeight: 500,
  },

  ".tweet-info-user span.secondary": {
    paddingLeft: "0.5rem",
    color: props.theme.secondaryColor,
  },

  ".tweet-body": {
    marginBottom: "0.75rem",
    wordBreak: "break-word",
  },

  ".tags": {
    display: "flex",
  },

  "a.body": {
    color: props.theme.accentColor,
  },

  "div.tweet-stats": {
    display: "inline-flex",
    flexWrap: "wrap",
    alignItems: "center",

    div: {
      color: props.theme.secondaryColor,
    },

    ".controls": {
      marginRight: "4rem",
    },

    svg: {
      marginRight: "0.5rem",
    },

    span: {
      display: "flex",
      alignItems: "center",
    },

    "span.comment": {
      svg: {
        position: "relative",
        top: "4px",
      },
    },
  },

  "@media screen and (max-width: 470px)": {
    "div.tweet-stats": {
      div: {
        marginRight: "1.5rem",
      },
      ".controls": {
        marginRight: "1.5rem",
      },
    },
  },

  "@media screen and (max-width: 430px)": {
    flexDirection: "column",

    ".username": {
      display: "none",
    },

    ".avatar": {
      display: "none",
    },

    ".tweet-info-user span.secondary": {
      paddingLeft: "0",
      paddingRight: "0.5rem",
    },
  },
}));

interface Props {
  tweet: Tweet;
}

export const ShowTweet: React.FC<Props> = ({ tweet }:Props) => {
  const {
    id,
    text,
    // tags,
    user,
    files,
    gif,
    nft,
    // isTweetMine,
    isRetweet,
    retweetsCount,
    reactions,
    commentsCount,
    tipsCount,
    createdAt,
  } = tweet;

  const { handleReaction } = useReaction({ tweetId: id });
  const handle = user && user.handle;

  const linkifyOptions = {
    className: "body",
    target: { url: "_blank" },
    formatHref: { hashtag: (href: any) => `explore?=${href.substring(1)}` },
  };
  return (
    <Wrapper>
      <Link to={`/${handle}`}>
        <UserAvatar className="avatar" avatar={user?.avatar as string} />
      </Link>

      <div className="tweet-info">
        <div className="tweet-info-user">
          <Link to={`/${handle}`}>
            <span className="username">{user && user.consumerName}</span>
            <span className="secondary">{`@${handle}`}</span>
          </Link>
          &nbsp;&nbsp;
          <Link to={`/${handle}/status/${id}`} className="secondary">
            {moment(setDate(createdAt)).fromNow()}
          </Link>
        </div>

        <Linkify options={linkifyOptions}>
          <p className="tweet-body">{text}</p>
        </Linkify>

        {gif && <VideoContainer gif={gif} />}

        {nft && <NFTTweet nftData={nft} />}

        {!!files.length && <ImageBox files={files} disablelightbox={false} />}

        {reactions.length > 0 && (
          <ReactionsList
            reactions={reactions}
            handleReaction={handleReaction}
            tweetId={id}
          />
        )}

        <div className="tweet-stats">
          <EmojiTweet handleReaction={handleReaction} />

          <div className="controls">
            <span className="comment">
              <Link to={`/${handle}/status/${id}`}>
                <CommentIcon />
                {commentsCount ? commentsCount : null}
              </Link>
            </span>
          </div>

          <div className="controls">
            <Retweet
              id={id}
              isRetweet={isRetweet}
              retweetsCount={retweetsCount}
            />
          </div>

          <>
            <TipCreator
              userPubKey={user?.publicAddress}
              tipAmount={tipsCount && parseInt(tipsCount) / LAMPORTS_PER_SOL}
              tweetId={id}
              userId={user?.id}
            />
          </>

          {/* <div>
            <span>{isTweetMine ? <DeleteTweet id={id} /> : null}</span>
          </div> */}
        </div>
      </div>
    </Wrapper>
  );
};
