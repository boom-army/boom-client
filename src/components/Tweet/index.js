export { DeleteTweet } from './DeleteTweet';
export { EmojiTweet } from './EmojiTweet';
export { MasterTweet } from './MasterTweet';
export { NewTweet } from './NewTweet';
export { ParentTweet } from './ParentTweet';
export { Retweet } from './Retweet';
export { ShowTweet } from './Tweet';
