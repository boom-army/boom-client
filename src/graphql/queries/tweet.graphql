query tweet($id: ID!) {
  tweet(id: $id) {
    ...TweetData
    masterTweets {
      ...TweetData
    }
  }
}

mutation newTweet(
  $text: String!
  $files: [String!]!
  $tags: [String!]!
  $mentions: [String!]!
  $gif: GifInput
  $nft: NFTInput
  $parentTweet: String
  $masterTweet: String
  $channel: String
) {
  newTweet(
    text: $text
    files: $files
    tags: $tags
    mentions: $mentions
    gif: $gif
    nft: $nft
    parentTweet: $parentTweet
    masterTweet: $masterTweet
    channel: $channel
  ) {
    id
    text
    tags
    mentions
    commentsCount
    createdAt
  }
}

mutation toggleRetweet($id: ID!) {
  toggleRetweet(id: $id)
}

mutation deleteTweet($id: ID!) {
  deleteTweet(id: $id) {
    id
  }
}
