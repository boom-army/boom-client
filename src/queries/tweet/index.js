import gql from "graphql-tag";

export const NEW_TWEET = gql`
  mutation newTweet(
    $text: String!
    $files: [String!]!
    $tags: [String!]!
    $mentions: [String!]!
    $gif: GifInput
    $nft: NFTInput
    $parentTweet: String
  ) {
    newTweet(
      text: $text
      files: $files
      tags: $tags
      mentions: $mentions
      gif: $gif
      nft: $nft
      parentTweet: $parentTweet
    ) {
      id
      text
      tags
      mentions
      commentsCount
      createdAt
    }
  }
`;

export const DELETE_TWEET = gql`
  mutation deleteTweet($id: ID!) {
    deleteTweet(id: $id) {
      id
    }
  }
`;

export const TOGGLE_RETWEET = gql`
  mutation toggleRetweet($id: ID!) {
    toggleRetweet(id: $id)
  }
`;

export const TOGGLE_REACTION = gql`
  mutation toggleReaction($id: ID!, $emojiId: String!, $skin: Int) {
    toggleReaction(id: $id, emojiId: $emojiId, skin: $skin)
  }
`;

const TWEET_FRAGMENT = gql`
  fragment subTweet on Tweet {
    id
    text
    tags
    mentions
    retweetsCount
    isRetweet
    createdAt
    parentTweet {
      id
    }
    user {
      id
      publicAddress
      consumerName
      handle
      avatar
    }
    files {
      id
      url
    }
    gif {
      id
      title
      fixedHeightUrl
      originalUrl
    }
    nft {
      id
      publicKey
      name
      symbol
      description
      sellerFeeBasisPoints
      externalUrl
      image
      attributes {
        traitType
        value
      }
      collection {
        name
        family
      }
      properties {
        files {
          uri
          type
        }
        category
        creators {
          address
          share
        }
      }
    }
    reactions {
      id
      emojiId
      skin
      isMine
      count
    }
  }
`;

export const TWEET = gql`
  query tweet($id: ID!) {
    tweet(id: $id) {
      ...subTweet
      childTweets {
        ...subTweet
        childTweets {
          ...subTweet
        }
      }
    }
  }
  ${TWEET_FRAGMENT}
`;
