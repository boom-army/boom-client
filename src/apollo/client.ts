import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          heroFeed: offsetLimitPagination(),
          getChannelById: offsetLimitPagination(),
          users: offsetLimitPagination(),
          searchTweets: offsetLimitPagination(),
          searchUser: offsetLimitPagination(),
          mentions: offsetLimitPagination(),
        },
      },
    },
  }),
  credentials: 'include',
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
    "Access-Control-Allow-Origin" : "*",
  },
  defaultOptions: {
    watchQuery: {
      context: async ({ req }: any) => {
        const token = req?.headers?.authorization || null;
        const user = token ? localStorage.getItem("user")?.toString() : null;
        return { user, isLoggedIn: !!token };
      },
    },
  },
});
