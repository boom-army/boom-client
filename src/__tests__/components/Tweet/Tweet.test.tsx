import React from "react";
import { ShowTweet } from "../../../components/Tweet/Tweet";
import renderer from "react-test-renderer";
import { MockedProvider } from "@apollo/client/testing";
// import { MentionsDocument, Tweet } from "../../../generated/graphql";
import { TWEET } from "../../../queries/tweet/index";

// export const __mocks__:any= [
//     {
//       request: {
//        query: TWEET,
//         variables: {
//           offset: 0,
//           limit: 10,
//         },
//       },
//       result: {
//         data: {
//           mentions:[]
//         ,
//         },
//         errors: "An error occurred",
//       },
//   }
//   ]

const Tweet: any = {
  id: "cl084xdy815761s0n5omoxoem0",
  text: "testing after changes in input converted to ts",
  tags: [],
  isTweetMine: true,
  commentsCount: 1,
  retweetsCount: 0,
  isRetweet: false,
  tipsCount: "0",
  createdAt: "1646139451616",
  parentTweet: null,
  files: [],
  gif: null,
  nft: null,
  user: {
    id: "ckzny1iv10031lmn568gj3b3n",
    publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
    avatar: "",
    handle: "shy-cloud-4965",
    consumerName: "shy-cloud-4965",
    __typename: "User",
  },
  reactions: [
    {
      id: "cl0c774p371312s0n533u0b29k",
      emojiId: "+1",
      skin: 1,
      isMine: true,
      count: 1,
      __typename: "Reaction",
    },
  ],
  __typename: "Tweet",
};

it("Jest works", () => {
  expect(true).toBeTruthy();
});

it("renders correctly", async () => {
  const component = renderer.create(
    //   <MockedProvider mocks={__mocks__} addTypename={false}>
    <ShowTweet tweet={Tweet} />
    //   </MockedProvider>,
  );

  await new Promise((resolve) => setTimeout(resolve, 0)); // wait for response

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  // expect(tree).toContain('An error occurred');

  // const tree = renderer.create(
  //   <Notifications/>
  // ).toJSON();
  // expect(tree).toMatchSnapshot();
});
