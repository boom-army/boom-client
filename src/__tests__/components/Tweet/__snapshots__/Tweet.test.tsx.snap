// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweet component snapshot testing ... 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-191cfdg"
      >
        <a
          href="/shy-cloud-4965"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar css-2s90m6-MuiAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </a>
        <div
          class="tweet-info"
        >
          <div
            class="tweet-info-user"
          >
            <a
              href="/shy-cloud-4965"
            >
              <span
                class="username"
              >
                shy-cloud-4965
              </span>
              <span
                class="secondary"
              >
                @shy-cloud-4965
              </span>
            </a>
              
            <a
              class="secondary"
              href="/shy-cloud-4965/status/cl084xdy815761s0n5omoxoem0"
            >
              16 days ago
            </a>
          </div>
          <p
            class="tweet-body"
          >
            testing after changes in input converted to ts
          </p>
          <div
            class="MuiBox-root css-1yt2crv"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-mui-internal-clone-element="true"
              style="border-width: 1px; padding: 4px 10px; min-width: auto; margin-right: 8px; color: rgb(101, 119, 134); border-color: #3f51b5; line-height: 1.2;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              >
                <span
                  aria-label="👍, +1, thumbsup"
                  class="emoji-mart-emoji"
                >
                  <span
                    style="width: 16px; height: 16px; display: inline-block; background-image: url(https://unpkg.com/emoji-datasource-apple@5.0.1/img/apple/sheets-256/64.png); background-size: 5700% 5700%; background-position: 23.214285714285715% 67.85714285714286%;"
                  />
                </span>
              </span>
              <p
                class="MuiTypography-root MuiTypography-body1 css-k5e2gc-MuiTypography-root"
              >
                1
              </p>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="tweet-stats"
          >
            <div
              class="css-687ye7"
            >
              <div
                class="css-s5gg9u"
              >
                <span
                  class="emoji-pick"
                >
                  <svg
                    fill="#637584"
                    height="18"
                    viewBox="0 0 18 18"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.6 2.4V0H14.4V2.4H12V3.6H14.4V6H15.6V3.6H18V2.4H15.6ZM5.09999 9H3.59999V10.5H5.09999V9ZM8.69998 9H10.2V10.5H8.69998V9ZM7.02895 13.8H6.77113C6.29273 13.8015 5.82489 13.6594 5.42823 13.3919C5.03156 13.1245 4.72436 12.7441 4.54641 12.3H3.25378L3.42861 12.737C3.69384 13.4064 4.15461 13.9803 4.75084 14.384C5.34708 14.7877 6.05111 15.0023 6.77113 15H7.02895C7.74897 15.0023 8.45299 14.7877 9.04922 14.384C9.64544 13.9803 10.1062 13.4064 10.3714 12.737L10.5463 12.3H9.25363C9.07569 12.7441 8.76849 13.1245 8.37183 13.3919C7.97517 13.6594 7.50734 13.8015 7.02895 13.8ZM0 11.1C0 7.29533 3.09532 4.2 6.9 4.2C10.7047 4.2 13.8 7.29533 13.8 11.1C13.8 14.9047 10.7047 18 6.9 18C3.09532 18 0 14.9047 0 11.1ZM1.2 11.1C1.2 14.243 3.75701 16.8 6.9 16.8C10.043 16.8 12.6 14.243 12.6 11.1C12.6 7.95702 10.043 5.4 6.9 5.4C3.75701 5.4 1.2 7.95702 1.2 11.1Z"
                    />
                  </svg>
                </span>
              </div>
            </div>
            <div
              class="controls"
            >
              <span
                class="comment"
              >
                <a
                  href="/shy-cloud-4965/status/cl084xdy815761s0n5omoxoem0"
                >
                  <svg
                    class="css-1csj31"
                    viewBox="0 0 24 24"
                  >
                    <g>
                      <path
                        d="M14.046 2.242l-4.148-.01h-.002c-4.374 0-7.8 3.427-7.8 7.802 0 4.098 3.186 7.206 7.465 7.37v3.828c0 .108.044.286.12.403.142.225.384.347.632.347.138 0 .277-.038.402-.118.264-.168 6.473-4.14 8.088-5.506 1.902-1.61 3.04-3.97 3.043-6.312v-.017c-.006-4.367-3.43-7.787-7.8-7.788zm3.787 12.972c-1.134.96-4.862 3.405-6.772 4.643V16.67c0-.414-.335-.75-.75-.75h-.396c-3.66 0-6.318-2.476-6.318-5.886 0-3.534 2.768-6.302 6.3-6.302l4.147.01h.002c3.532 0 6.3 2.766 6.302 6.296-.003 1.91-.942 3.844-2.514 5.176z"
                      />
                    </g>
                  </svg>
                  1
                </a>
              </span>
            </div>
            <div
              class="controls"
            >
              <span>
                <svg
                  class="css-1csj31"
                  loading=""
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M23.77 15.67c-.292-.293-.767-.293-1.06 0l-2.22 2.22V7.65c0-2.068-1.683-3.75-3.75-3.75h-5.85c-.414 0-.75.336-.75.75s.336.75.75.75h5.85c1.24 0 2.25 1.01 2.25 2.25v10.24l-2.22-2.22c-.293-.293-.768-.293-1.06 0s-.294.768 0 1.06l3.5 3.5c.145.147.337.22.53.22s.383-.072.53-.22l3.5-3.5c.294-.292.294-.767 0-1.06zm-10.66 3.28H7.26c-1.24 0-2.25-1.01-2.25-2.25V6.46l2.22 2.22c.148.147.34.22.532.22s.384-.073.53-.22c.293-.293.293-.768 0-1.06l-3.5-3.5c-.293-.294-.768-.294-1.06 0l-3.5 3.5c-.294.292-.294.767 0 1.06s.767.293 1.06 0l2.22-2.22V16.7c0 2.068 1.683 3.75 3.75 3.75h5.85c.414 0 .75-.336.75-.75s-.337-.75-.75-.75z"
                    />
                  </g>
                </svg>
              </span>
            </div>
            <span
              class="css-1e2zaxw"
            >
              <svg
                class="css-1csj31"
                viewBox="0 0 24 24"
              >
                <path
                  clip-rule="evenodd"
                  d="M20.9518 9.4744C21.3776 11.353 21.2062 13.3184 20.4618 15.0941L19.0322 14.266L19.2216 19.6957L24 17.1456L22.4603 16.254C24.8616 10.9233 22.8883 4.53067 17.7369 1.54514C16.0006 0.537631 14.0315 0.00488574 12.026 3.34445e-05C10.0206 -0.00481885 8.04895 0.518405 6.30774 1.51749L7.44592 3.51822C9.11194 2.56211 11.0345 2.1543 12.9435 2.35205C14.8523 2.54984 16.6515 3.34326 18.088 4.62082C19.5244 5.89835 20.526 7.59577 20.9518 9.4744ZM3.04944 13.5656C2.62366 11.6875 2.79431 9.72245 3.53821 7.94702L4.96777 8.77518L4.77844 3.34543L0 5.8956L1.54077 6.78598C-0.860474 12.1166 1.11292 18.5093 6.26416 21.4949C8.00049 22.5024 9.9696 23.0351 11.9751 23.04C13.9806 23.0448 15.9523 22.5216 17.6934 21.5225L16.5553 19.5218C14.8893 20.4779 12.9666 20.8857 11.0577 20.6879C9.1488 20.4902 7.34961 19.6967 5.91321 18.4192C4.47668 17.1417 3.47522 15.4442 3.04944 13.5656ZM12.176 8.75968C11.2129 8.75968 10.1024 8.88932 10.1024 9.6485C10.1024 10.3847 11.1198 10.6636 12.3204 10.9928C14.0487 11.4667 16.157 12.0447 16.157 14.2408C16.157 15.2958 15.62 16.8884 12.6204 17.1478L12.9166 19.222H10.9353L11.2318 17.1667C9.78711 17.0923 8.58423 16.7777 7.99109 16.4258L8.30627 14.0739C8.9353 14.6669 10.1578 15.2403 11.787 15.2403C12.991 15.2403 13.6755 14.9817 13.6755 14.2589C13.6755 13.4128 12.5487 13.0673 11.2856 12.68C9.67517 12.1861 7.84326 11.6244 7.84326 9.8707C7.84326 8.62998 8.67676 7.14872 11.2318 6.90836L10.9353 4.77872H12.9166L12.6204 6.90836C13.8427 6.98188 15.0459 7.3338 15.5646 7.63034L15.3613 9.72284C14.3242 9.18512 13.0837 8.75968 12.176 8.75968Z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-191cfdg"
    >
      <a
        href="/shy-cloud-4965"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar css-2s90m6-MuiAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </a>
      <div
        class="tweet-info"
      >
        <div
          class="tweet-info-user"
        >
          <a
            href="/shy-cloud-4965"
          >
            <span
              class="username"
            >
              shy-cloud-4965
            </span>
            <span
              class="secondary"
            >
              @shy-cloud-4965
            </span>
          </a>
            
          <a
            class="secondary"
            href="/shy-cloud-4965/status/cl084xdy815761s0n5omoxoem0"
          >
            16 days ago
          </a>
        </div>
        <p
          class="tweet-body"
        >
          testing after changes in input converted to ts
        </p>
        <div
          class="MuiBox-root css-1yt2crv"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            data-mui-internal-clone-element="true"
            style="border-width: 1px; padding: 4px 10px; min-width: auto; margin-right: 8px; color: rgb(101, 119, 134); border-color: #3f51b5; line-height: 1.2;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <span
                aria-label="👍, +1, thumbsup"
                class="emoji-mart-emoji"
              >
                <span
                  style="width: 16px; height: 16px; display: inline-block; background-image: url(https://unpkg.com/emoji-datasource-apple@5.0.1/img/apple/sheets-256/64.png); background-size: 5700% 5700%; background-position: 23.214285714285715% 67.85714285714286%;"
                />
              </span>
            </span>
            <p
              class="MuiTypography-root MuiTypography-body1 css-k5e2gc-MuiTypography-root"
            >
              1
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="tweet-stats"
        >
          <div
            class="css-687ye7"
          >
            <div
              class="css-s5gg9u"
            >
              <span
                class="emoji-pick"
              >
                <svg
                  fill="#637584"
                  height="18"
                  viewBox="0 0 18 18"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15.6 2.4V0H14.4V2.4H12V3.6H14.4V6H15.6V3.6H18V2.4H15.6ZM5.09999 9H3.59999V10.5H5.09999V9ZM8.69998 9H10.2V10.5H8.69998V9ZM7.02895 13.8H6.77113C6.29273 13.8015 5.82489 13.6594 5.42823 13.3919C5.03156 13.1245 4.72436 12.7441 4.54641 12.3H3.25378L3.42861 12.737C3.69384 13.4064 4.15461 13.9803 4.75084 14.384C5.34708 14.7877 6.05111 15.0023 6.77113 15H7.02895C7.74897 15.0023 8.45299 14.7877 9.04922 14.384C9.64544 13.9803 10.1062 13.4064 10.3714 12.737L10.5463 12.3H9.25363C9.07569 12.7441 8.76849 13.1245 8.37183 13.3919C7.97517 13.6594 7.50734 13.8015 7.02895 13.8ZM0 11.1C0 7.29533 3.09532 4.2 6.9 4.2C10.7047 4.2 13.8 7.29533 13.8 11.1C13.8 14.9047 10.7047 18 6.9 18C3.09532 18 0 14.9047 0 11.1ZM1.2 11.1C1.2 14.243 3.75701 16.8 6.9 16.8C10.043 16.8 12.6 14.243 12.6 11.1C12.6 7.95702 10.043 5.4 6.9 5.4C3.75701 5.4 1.2 7.95702 1.2 11.1Z"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            class="controls"
          >
            <span
              class="comment"
            >
              <a
                href="/shy-cloud-4965/status/cl084xdy815761s0n5omoxoem0"
              >
                <svg
                  class="css-1csj31"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M14.046 2.242l-4.148-.01h-.002c-4.374 0-7.8 3.427-7.8 7.802 0 4.098 3.186 7.206 7.465 7.37v3.828c0 .108.044.286.12.403.142.225.384.347.632.347.138 0 .277-.038.402-.118.264-.168 6.473-4.14 8.088-5.506 1.902-1.61 3.04-3.97 3.043-6.312v-.017c-.006-4.367-3.43-7.787-7.8-7.788zm3.787 12.972c-1.134.96-4.862 3.405-6.772 4.643V16.67c0-.414-.335-.75-.75-.75h-.396c-3.66 0-6.318-2.476-6.318-5.886 0-3.534 2.768-6.302 6.3-6.302l4.147.01h.002c3.532 0 6.3 2.766 6.302 6.296-.003 1.91-.942 3.844-2.514 5.176z"
                    />
                  </g>
                </svg>
                1
              </a>
            </span>
          </div>
          <div
            class="controls"
          >
            <span>
              <svg
                class="css-1csj31"
                loading=""
                viewBox="0 0 24 24"
              >
                <g>
                  <path
                    d="M23.77 15.67c-.292-.293-.767-.293-1.06 0l-2.22 2.22V7.65c0-2.068-1.683-3.75-3.75-3.75h-5.85c-.414 0-.75.336-.75.75s.336.75.75.75h5.85c1.24 0 2.25 1.01 2.25 2.25v10.24l-2.22-2.22c-.293-.293-.768-.293-1.06 0s-.294.768 0 1.06l3.5 3.5c.145.147.337.22.53.22s.383-.072.53-.22l3.5-3.5c.294-.292.294-.767 0-1.06zm-10.66 3.28H7.26c-1.24 0-2.25-1.01-2.25-2.25V6.46l2.22 2.22c.148.147.34.22.532.22s.384-.073.53-.22c.293-.293.293-.768 0-1.06l-3.5-3.5c-.293-.294-.768-.294-1.06 0l-3.5 3.5c-.294.292-.294.767 0 1.06s.767.293 1.06 0l2.22-2.22V16.7c0 2.068 1.683 3.75 3.75 3.75h5.85c.414 0 .75-.336.75-.75s-.337-.75-.75-.75z"
                  />
                </g>
              </svg>
            </span>
          </div>
          <span
            class="css-1e2zaxw"
          >
            <svg
              class="css-1csj31"
              viewBox="0 0 24 24"
            >
              <path
                clip-rule="evenodd"
                d="M20.9518 9.4744C21.3776 11.353 21.2062 13.3184 20.4618 15.0941L19.0322 14.266L19.2216 19.6957L24 17.1456L22.4603 16.254C24.8616 10.9233 22.8883 4.53067 17.7369 1.54514C16.0006 0.537631 14.0315 0.00488574 12.026 3.34445e-05C10.0206 -0.00481885 8.04895 0.518405 6.30774 1.51749L7.44592 3.51822C9.11194 2.56211 11.0345 2.1543 12.9435 2.35205C14.8523 2.54984 16.6515 3.34326 18.088 4.62082C19.5244 5.89835 20.526 7.59577 20.9518 9.4744ZM3.04944 13.5656C2.62366 11.6875 2.79431 9.72245 3.53821 7.94702L4.96777 8.77518L4.77844 3.34543L0 5.8956L1.54077 6.78598C-0.860474 12.1166 1.11292 18.5093 6.26416 21.4949C8.00049 22.5024 9.9696 23.0351 11.9751 23.04C13.9806 23.0448 15.9523 22.5216 17.6934 21.5225L16.5553 19.5218C14.8893 20.4779 12.9666 20.8857 11.0577 20.6879C9.1488 20.4902 7.34961 19.6967 5.91321 18.4192C4.47668 17.1417 3.47522 15.4442 3.04944 13.5656ZM12.176 8.75968C11.2129 8.75968 10.1024 8.88932 10.1024 9.6485C10.1024 10.3847 11.1198 10.6636 12.3204 10.9928C14.0487 11.4667 16.157 12.0447 16.157 14.2408C16.157 15.2958 15.62 16.8884 12.6204 17.1478L12.9166 19.222H10.9353L11.2318 17.1667C9.78711 17.0923 8.58423 16.7777 7.99109 16.4258L8.30627 14.0739C8.9353 14.6669 10.1578 15.2403 11.787 15.2403C12.991 15.2403 13.6755 14.9817 13.6755 14.2589C13.6755 13.4128 12.5487 13.0673 11.2856 12.68C9.67517 12.1861 7.84326 11.6244 7.84326 9.8707C7.84326 8.62998 8.67676 7.14872 11.2318 6.90836L10.9353 4.77872H12.9166L12.6204 6.90836C13.8427 6.98188 15.0459 7.3338 15.5646 7.63034L15.3613 9.72284C14.3242 9.18512 13.0837 8.75968 12.176 8.75968Z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
